src/utils.jl

```diff
@@ -51,14 +51,19 @@ checking if `expr` has finished executing (short for polling interval).
 """
 macro timeout(t, expr, then, pollint=0.01)
     return quote
-        tm = Float64($(esc(t)))
-        start = time()
-        tsk = @async $(esc(expr))
-        while !istaskdone(tsk) && (time() - start < tm)
-            sleep($pollint)
+        if $(esc(t)) == Inf
+            $(esc(expr))
+        else
+            tm = Float64($(esc(t)))
+            start = time()
+            tsk = @async $(esc(expr))
+            yield()
+            while !istaskdone(tsk) && (time() - start < tm)
+                sleep($pollint)
+            end
+            istaskdone(tsk) || $(esc(then))
+            wait(tsk)
```

Comment:
Can you remind me why the change to `wait` here?

Reply:
Because `wait` rethrows the task exception if needed, whereas the current code ends up at `no method matching length(::Base.UVError)`

From #113 ...

The [`timeout` macro](https://github.com/JuliaWeb/HTTP.jl/blob/master/src/utils.jl#L52) executes `expr` in a new task, waits for a timeout or `istaskdone`, then returns `task.result`. In the case where `expr` throws and exception, the exception is returned as a value when it should be re-thrown.

The result is:
```
ERROR: MethodError: no method matching length(::Base.UVError)
 [1] processresponse!(::HTTP.Client, ::HTTP.Connection{MbedTLS.SSLContext}, ::HTTP.Response, ::String, ::HTTP.Method, ::Task, ::Bool, ::Float64, ::Bool, ::Bool)
at /Users/sam/.julia/v0.6/HTTP/src/client.jl:287
```